src/ExpressionParser.php

```diff
@@ -432,6 +432,52 @@ public function parseMappingExpression()
         return $node;
     }
 
+    /*
+     * Unlike its sibling methods above, method requires a TokenStream parameter.
+     * 
+     * The reason is that calling `$this->parser->getStream()` (as in `parseMappingExpression()`)
+     * only works _while_ `parser->parse()` is running, because that method sets and unsets the $stream property.
+     * 
+     * The upside is that this method is fully pure, since it doesn't depend on the state of $this.
+     */
+    public function parseNameToStringMapping(TokenStream $stream, bool $allowOptionalKeys): ArrayExpression
+    {
+        $stream->expect(Token::PUNCTUATION_TYPE, '{', 'A mapping element was expected');
```

Comment:
I think this should be like this, to follow the rest of the ExpressionParser

```suggestion
    public function parseNameToStringMapping(bool $allowOptionalKeys): ArrayExpression
    {
        $stream = $this->parser->getStream();
        $stream->expect(Token::PUNCTUATION_TYPE, '{', 'A mapping element was expected');
```